WeatherApp/WeatherApp/App/Data/Repositories/DataRepository.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+
+protocol DataRepositoryProtocol {
+
+    func storeCities(cities: [City])
+    func getCities() -> [City]
+
+}
+
+class DataRepository: DataRepositoryProtocol {
+
+    private let defaultCities: [City] = [
+        City(name: "Zagreb"),
+        City(name: "Paris"),
+        City(name: "New York"),
+        City(name: "Tokyo"),
+        City(name: "London"),
+        City(name: "Los Angeles")]
+
+    let dataService: DataServiceProtocol
+
+    init(dataService: DataServiceProtocol) {
+        self.dataService = dataService
+    }
+
+    func storeCities(cities: [City]) {
+        dataService.storeCities(cities: cities)
+    }
+
+    func getCities() -> [City] {
+        var cities = dataService.getCities()
+
+        if cities.isEmpty {
+            cities = defaultCities
+        }
+
+        return cities
+    }
+
+}
```

Comment:
Not sure if I would go with this approach. This data is only needed to store the weather for those locations.

I would make this a part of the WeatherRepository and treat it like "ignorable data". At the end you want a call `getWeatherForSavedLocations` that will use the stored location in conjuction with the weather data you get from the endpoint. In the end you will want that kind of connection in the database

Reply:
This is addressed in the newest PR when Realm is being used.